btree/btree.c

```diff
@@ -4530,88 +4534,51 @@ static void fkvv_del_emb_ind(const struct nd *node, int idx)
 {
 	struct fkvv_head    *h            = fkvv_data(node);
 	struct fkvv_dir_rec *dir          = fkvv_dir_get(&node->n_addr);
-	int                  free_farg_1  = -1;
-	int                  free_farg_2  = -1;
+	struct fkvv_dir_rec  dir_ent      = dir[idx];
+	int                  freed_ent;
 	int                  i;
-
-	for (i = h->fkvv_used; i < h->fkvv_dir_entries; i++) {
-		if (i == idx)
-			continue;
-		/**
-		 * Find if there exist any contiguous left key and right value
-		 * free fragment which can be merged  dir_entry fragment.
-		 */
-		if (free_farg_1 == -1 &&
-		    dir[idx].key_offset == dir[i].key_offset +
-		    			   dir[i].key_size  &&
-		    dir[i].val_offset == dir[idx].val_offset +
-		    			 dir[idx].alloc_val_size) {
-			free_farg_1 = i;
-		}
-
-		/**
-		 * Find if there exist any contiguous right key and left value
-		 * free fragment which can be merged  dir[idx] fragment.
-		 */
-		if (free_farg_2 == -1 &&
-		    dir[i].key_offset == dir[idx].key_offset +
-		    			 dir[idx].key_size &&
-		    dir[idx].val_offset == dir[i].val_offset +
-		    			   dir[i].alloc_val_size) {
-			free_farg_2 = i;
-		}
-	}
-
-	if (free_farg_1 != -1) {
-		dir[free_farg_1].key_size       += dir[idx].key_size;
-		dir[free_farg_1].alloc_val_size += dir[idx].alloc_val_size;
-		dir[free_farg_1].val_offset     -= dir[idx].alloc_val_size;
-		dir[idx] = dir[free_farg_1];
+	int                  bytes_to_move;
+	h->fkvv_used--;
+	if (h->fkvv_used == 0) {
+		fkvv_dir_init(&node->n_addr);
```

Comment:
By returning here you are skipping the chance to merge this entry with some other entry. You do not want to do this.

Reply:
No. fkvv_dir_init() will also reset h->fkvv_dir_entries to 1 and add 1 entry of free fragment.